machine/dispatch.py

```diff
@@ -40,8 +40,9 @@ def handle_message(self, client, event):
         # Handle message listeners
         # Handle message subtype 'message_changed' to allow the bot to respond to edits
         if 'subtype' in event and event['subtype'] == 'message_changed':
+            channel = event.get('channel', '')
             event = event['message']
-            event['channel'] = event.get('channel', '')
+            event['channel'] = channel
```

Comment:
I don't understand this change. It doesn't seem to do anything different functionally?

Reply:
When the `event` comes in, it looks something like `event = {'subtype': 'message_changed', 'message': {...}, 'channel': 'abcdefg'}` where `event['message']` has no field named `channel`, as the field `channel` is at the top level. So by storing the channel before overwriting `event` with the `message` contents, I'm saving the `channel` and adding it to the (new) `event` object. Since that is used later on I was getting an error when it wasn't saved as the `message` dictionary doesn't contain the `channel` when the subtype is `message_changed`.
Hopefully that makes more sense 😅

Reply:
aaah now I see. Makes sense!